src/aks-preview/azext_aks_preview/addonconfiguration.py

```diff
@@ -497,16 +498,39 @@ def sanitize_loganalytics_ws_resource_id(workspace_resource_id):
     return workspace_resource_id
 
 
-def ensure_container_insights_for_monitoring(cmd,
-                                             addon,
-                                             cluster_subscription,
-                                             cluster_resource_group_name,
-                                             cluster_name,
-                                             cluster_region,
-                                             remove_monitoring=False,
-                                             aad_route=False,
-                                             create_dcr=False,
-                                             create_dcra=False):
+def get_existing_container_insights_extension_dcr_tags(cmd, dcr_url):
```

Comment:
I would say it's better to import the code from azure-cli to align them. Otherwise, there will always be more or less differences between the two sides. One example would be `ensure_default_log_analytics_workspace_for_monitoring`.

At present, a large part of the implementation of the `aks create/update` commands reuse the code in azure-cli.

Reply:
> I would say it's better to import the code from azure-cli to align them. Otherwise, there will always be more or less differences between the two sides. One example would be `ensure_default_log_analytics_workspace_for_monitoring`.
> 
> At present, a large part of the implementation of the `aks create/update` commands reuse the code in azure-cli.

> imported the code from azure-cli and remove the code from aks-preview code.